File: pm_core/src/infrastructure/store/mapper/priority_mapper.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package mapper

import (
	"github.com/serp/pm-core/src/core/domain/entity"
	"github.com/serp/pm-core/src/infrastructure/store/model"
)

type PriorityMapper struct {
}

func NewPriorityMapper() *PriorityMapper {
	return &PriorityMapper{}
}

func (m *PriorityMapper) ToEntity(mdl *model.PriorityModel) *entity.PriorityEntity {
	if mdl == nil {
		return nil
	}
	return &entity.PriorityEntity{
		BaseEntity: entity.BaseEntity{
			ID:        mdl.ID,
			CreatedAt: mdl.CreatedAt.UnixMilli(),
			CreatedBy: mdl.CreatedBy,
			UpdatedAt: mdl.UpdatedAt.UnixMilli(),
			UpdatedBy: mdl.UpdatedBy,
		},
		TenantID:    mdl.TenantID,
		Name:        mdl.Name,
		Description: mdl.Description,
		IconUrl:     mdl.IconUrl,
		Color:       mdl.Color,
		Sequence:    mdl.Sequence,
		IsSystem:    mdl.IsSystem,
	}
}

func (m *PriorityMapper) ToModel(e *entity.PriorityEntity) *model.PriorityModel {
	if e == nil {
		return nil
	}
	return &model.PriorityModel{
		BaseModel: model.BaseModel{
			ID:        e.ID,
			CreatedBy: e.CreatedBy,
			UpdatedBy: e.UpdatedBy,
		},
		TenantID:    e.TenantID,
		Name:        e.Name,
		Description: e.Description,
		IconUrl:     e.IconUrl,
		Color:       e.Color,
		Sequence:    e.Sequence,
		IsSystem:    e.IsSystem,
	}
}

func (m *PriorityMapper) ToEntities(models []*model.PriorityModel) []*entity.PriorityEntity {
	if models == nil {
		return nil
	}
	entities := make([]*entity.PriorityEntity, 0, len(models))
	for _, mdl := range models {
		if e := m.ToEntity(mdl); e != nil {
			entities = append(entities, e)
		}
	}

	return entities
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/mapper/PriorityMapper.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.mapper;

import org.springframework.stereotype.Component;
import serp.project.pmcore.core.domain.entity.PriorityEntity;
import serp.project.pmcore.infrastructure.store.model.PriorityModel;

import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

@Component
public class PriorityMapper extends BaseMapper {

    public PriorityEntity toEntity(PriorityModel model) {
        if (model == null) { return null; }
        return PriorityEntity.builder()
                .id(model.getId())
                .tenantId(model.getTenantId())
                .name(model.getName())
                .description(model.getDescription())
                .iconUrl(model.getIconUrl())
                .color(model.getColor())
                .sequence(model.getSequence())
                .isSystem(Boolean.TRUE.equals(model.getIsSystem()))
                .createdAt(localDateTimeToLong(model.getCreatedAt()))
                .createdBy(model.getCreatedBy())
                .updatedAt(localDateTimeToLong(model.getUpdatedAt()))
                .updatedBy(model.getUpdatedBy())
                .build();
    }

    public PriorityModel toModel(PriorityEntity entity) {
        if (entity == null) { return null; }
        return PriorityModel.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .name(entity.getName())
                .description(entity.getDescription())
                .iconUrl(entity.getIconUrl())
                .color(entity.getColor())
                .sequence(entity.getSequence())
                .isSystem(entity.isSystem())
                .createdAt(longToLocalDateTime(entity.getCreatedAt()))
                .createdBy(entity.getCreatedBy())
                .updatedAt(longToLocalDateTime(entity.getUpdatedAt()))
                .updatedBy(entity.getUpdatedBy())
                .build();
    }

    public List<PriorityEntity> toEntities(List<PriorityModel> models) {
        if (models == null) { return Collections.emptyList(); }
        return models.stream().map(this::toEntity).collect(Collectors.toList());
    }
}
